core/src/ast_builder/select/filter.rs

```diff
@@ -100,7 +102,17 @@ impl<'a> FilterNode<'a> {
 impl<'a> Prebuild for FilterNode<'a> {
     fn prebuild(self) -> Result<NodeData> {
         let mut select_data = self.prev_node.prebuild()?;
-        select_data.filter = Some(self.filter_expr.try_into()?);
+        match select_data {
+            NodeData::Select(ref mut select_data) => {
+                select_data.filter = Some(self.filter_expr.try_into()?)
+            }
+            NodeData::Values(_) => {
+                return Err(
+                    AstBuilderError::UnreachableNode("ValuesData -> FilterNode".to_owned()).into(),
+                )
```

Comment:
It looks because the current structure is not the ideal one. I think that we can find the better shape for this.

![image](https://user-images.githubusercontent.com/2025065/207334683-03c2b482-1f13-4f77-b52d-007360f92c7a.png)

Now can we consider to make the shape similar with the `ast::Query` that shared `order_by`, `limit` and `offset` and split using `SetExpr`. the same approach may work.